### PR TITLE
TST: Add tests for if_exists keyword argument in df.to_gbq

### LINKED ISSUE #29598: pandas.DataFrame.to_gbq missing method of writting
BigQuery support 3 different types of load methods:
WRITE_EMPTY, WRITE_TRUNCATE, WRITE_APPEND
https://cloud.google.com/bigquery/docs/reference/auditlogs/rest/Shared.Types/WriteDisposition

this parameter needs to be controlled by loading job.

### PR DESCRIPTION
**Pull request by charlesdong1991**
- [x] closes #29598
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -891,7 +891,7 @@ Other
 - Fixed :class:`IntegerArray` returning ``inf`` rather than ``NaN`` for operations dividing by 0 (:issue:`27398`)
 - Fixed ``pow`` operations for :class:`IntegerArray` when the other value is ``0`` or ``1`` (:issue:`29997`)
 - Bug in :meth:`Series.count` raises if use_inf_as_na is enabled (:issue:`29478`)
-
+- Add tests for keyword argument ``if_exists`` in :meth:`DataFrame.to_gbq` (:issue:`29598`)
```
Comment (WillAyd) [inline doc/source/whatsnew/v1.0.0.rst]: No note needed as this isn't a user-facing change

**Review by WillAyd — CHANGES_REQUESTED** [review]
Minor comment otherwise ping on green

File: doc/source/whatsnew/v1.0.0.rst (hunk 1 above)
Comment (charlesdong1991) [reply doc/source/whatsnew/v1.0.0.rst]: ping 
@WillAyd

**Comment by charlesdong1991** [comment]
ping
@WillAyd

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @charlesdong1991

**Comment by alimcmaster1** [comment]
Potentially might neater to parameterize the test above `test_roundtrip` since the pattern is very similar?

**Comment by charlesdong1991** [comment]
thanks for your comment @alimcmaster1 

sorry that i just noticed it. I quickly looked at the `test_round_trip` method, but I am not sure if it is need to parametrize. Could you clarify for which part you think to parametrize?

thanks!

**Pull request merged**